cmd/security-agent/main_windows.go

```diff
@@ -154,7 +154,7 @@ func (s *service) Run(svcctx context.Context) error {
 			},
 		),
 		fx.Provide(func(config config.Component) status.HeaderInformationProvider {
```

Comment:
Why are we doing this in the first place ? Do we know why the security-agent needs a `status.HeaderInformationProvider` ?

Splitting the status from the component it represent does reduce the dependencies but doesn't make sens from a design POV. The issue here might be that the security-agent should not depend on the host metadata payload at all.

Reply:
It's used to get host info in the status output 🤷‍♂️

Reply:
Thinking about it, I think I agree with Maxime. This provider doesn't make much sense in that it's not even fetching the hostname from the core agent as it should. Let me take a look at what it would change to remove it completely

Reply:
I think it would just remove the section in the status output, nothing else should use it

Reply:
yes and the info cannot be trusted anyway. Let me open a PR to remove it

Reply:
@hush-hush do you want me to close this PR or not ?
I still think splitting the status makes sense considering it's a separate piece of logic from everything else in that package

Reply:
I'd rather we close the PR. If we want a dedicated logic block for the status page we would do it outside of the metadata components. This also goes against the guideline we highlighted for components where we try to consider them as black box and avoid importing them as package.
If we really want to extract the status we should move it as a stateless helper in pkg for example.